Dockerfile

```diff
@@ -0,0 +1,21 @@
+FROM ruby:2.3.1
+RUN apt-get update -qq && apt-get upgrade -y
+
+RUN apt-get install -y build-essential nodejs && apt-get clean
+
+ENV GOVUK_APP_NAME frontend
+ENV GOVUK_ASSET_ROOT http://assets-origin.dev.gov.uk
+ENV PORT 3005
+ENV RAILS_ENV development
+
+ENV APP_HOME /app
+RUN mkdir $APP_HOME
+
+WORKDIR $APP_HOME
+ADD Gemfile* $APP_HOME/
+ADD .ruby-version $APP_HOME/
+RUN bundle install
```

Comment:
Is this so that a Docker snapshot is created after bundling so that the container need not be re-created if the gems have remained the same?

I was puzzled why these files are picked out specifically rather than `ADD . $APP_HOME` then bundling?

Reply:
To be honest, I just pinched this from whoever did government-frontend. If it were me, I probably would just have done the below, but I assumed whoever built the government-frontend one did it a reason:

```
COPY . .
RUN bundle install
```

Reply:
I'm pretty sure this was been done to support caching, so if the `Gemfile*` or `.ruby-version` hasn't changed, it won't need to run `bundle install` again.

Reply:
Sure, makes sense.